relay-server/src/services/buffer/mod.rs

```diff
@@ -306,9 +306,9 @@ impl EnvelopeBufferService {
                     "EnvelopeBufferService: received project not ready message for project key {}",
                     &project_key
                 );
-                buffer.mark_ready(&project_key, false);
                 relay_statsd::metric!(counter(RelayCounters::BufferEnvelopesReturned) += 1);
                 self.push(buffer, envelope).await;
+                buffer.mark_ready(&project_key, false);
```

Comment:
This was a pre-existing bug: Calling mark_ready on an empty stack has no effect, so we should call it _after_ putting back the envelope.